### PR TITLE
CIP-0008: Allow for signing with stake key directly

### PR DESCRIPTION
**Pull request by thaddeusdiamond**
CIP-0008: Allow for signing with stake key directly
In the browser, if a user specifies the stake key as part of CIP-0030
the message returned in JSON has the COSE key as the staking key.  The
verify() method correctly detects this, but the sign() method does not.
This change detects if a stake SigningKey type was passed in and if so,
sign with a stake key.

Sample behavior:

[00:28:29 01/24 thaddeuss-mbp] hacker:~/workspaces/wildtangz/cardano-vending-machine$ python3
Python 3.8.1 (v3.8.1:1b293b6006, Dec 18 2019, 14:08:53)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pycardano
>>> skey = pycardano.key.StakeSigningKey.load('/var/folders/qz/ldnst4y96_z436s0ptw0vyp40000gn/T/cardano-vm-test-eobtvu6g/buyers/buyer.stake.skey')
>>> sign_addr = pycardano.cip.cip8.verify(pycardano.cip.cip8.sign('hello', skey, attach_cose_key=False, network=pycardano.network.Network.TESTNET))['signing_address']
>>> sign_addr
stake_test1uq5xgs974dna05tjc5pr8sfq5cc4hzludaz7z4e4vpx6p0gkrzzmd

The old behavior resulted in output of 'addr_test1vq5xgs974dna05tjc5pr8sfq5cc4hzludaz7z4e4vpx6p0gktu6v8'

[ Documentation: pydoc updates ]
[ Testing: python -m build, manual test in CLI ]
[ Code Review: https://github.com/Python-Cardano/pycardano/pull/154 ]

### CODE REVIEW

**Comment by thaddeusdiamond** [comment]
Let me know if the versioning update I put in here is not correct I am not certain on your version conventions.

**Comment by thaddeusdiamond** [comment]
Tagging @cffls for review when you have a moment.

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pycardano"
-version = "0.7.2"
+version = "0.7.3"
```
Comment (cffls) [inline pyproject.toml]: We usually update version in a different workflow outside of feature PR. Let's remove this line of change.

File: pycardano/cip/cip8.py [hunk 2]
```diff
@@ -43,11 +47,16 @@ def sign(
     # derive the verification key
     verification_key = VerificationKey.from_signing_key(signing_key)
 
+    if isinstance(signing_key, StakeSigningKey) or isinstance(signing_key, StakeExtendedSigningKey):
+        address = Address(payment_part=None, staking_part=verification_key.hash(), network=network)
+    else:
+        address = Address(payment_part=verification_key.hash(), staking_part=None, network=network)
```
Comment (cffls) [inline pycardano/cip/cip8.py]: This change looks good to me. Could you add a unit test for it?

**Review by cffls — COMMENTED** [review]
Hi @thaddeusdiamond , thank you so much for your contribution, and apologies for the delayed response. The PR looks good to me. Also, it would be great if you can add a unit test for it. Thank you!

File: pycardano/cip/cip8.py (hunk 2 above)
Comment (thaddeusdiamond) [reply pycardano/cip/cip8.py]: Done.

File: pyproject.toml (hunk 1 above)
Comment (thaddeusdiamond) [reply pyproject.toml]: Done.

**Comment by thaddeusdiamond** [comment]
No worries on delay just checking in! See the new commit for the unit tests and version removal.

**Comment by thaddeusdiamond** [comment]
Didn't realize you used black, that's why tests failed.  Ran black and repushed commit.

**Review by cffls — APPROVED** [review]
LGTM. Thank you for adding tests and fixing the code format!

**Pull request merged**